src/render/draw_line.js

```diff
@@ -18,17 +18,16 @@ export default function drawLine(painter: Painter, sourceCache: SourceCache, lay
     if (opacity.constantOr(1) === 0) return;
 
     const context = painter.context;
-
     context.setDepthMode(painter.depthModeForSublayer(0, DepthMode.ReadOnly));
     context.setColorMode(painter.colorModeForRenderPass());
 
+    const linePattern = layer.paint.get('line-pattern');
     const programId =
         layer.paint.get('line-dasharray') ? 'lineSDF' :
-        layer.paint.get('line-pattern') ? 'linePattern' : 'line';
+        linePattern && linePattern.constantOr((1: any)) ? 'linePattern' : 'line';
```

Comment:
open to better ways of doing this check – basically if `line-pattern` is undefined, `constantOr(1)` will return undefined, but `linePattern` itself is no longer falsy if `line-pattern` is undefined because it is now a `PossiblyEvaluatedPropertyValue`

Reply:
I'm okay with this as-is, since it's analogous to how we do other such checks.

Reply:
Is there a way to eliminate the `any` cast?